public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.test.ts

```diff
@@ -79,254 +62,527 @@ describe('CloudWatchLogsQueryRunner', () => {
     });
   });
 
-  describe('logs query', () => {
-    beforeEach(() => {
-      jest.spyOn(rxjsUtils, 'increasingInterval').mockImplementation(() => interval(100));
-    });
+  describe('handleLogQueries', () => {
+    it('should request to start each query and then request to get the query results', async () => {
+      const { runner } = setupMockedLogsQueryRunner();
 
-    it('should stop querying when timed out', async () => {
-      const { runner, queryMock } = setupMockedLogsQueryRunner();
-      const fakeFrames = genMockFrames(20);
-      const initialRecordsMatched = fakeFrames[0].meta!.stats!.find(
-        (stat) => stat.displayName === 'Records scanned'
-      )!.value!;
-      for (let i = 1; i < 4; i++) {
-        fakeFrames[i].meta!.stats = [
-          {
-            displayName: 'Records scanned',
-            value: initialRecordsMatched,
-          },
-        ];
-      }
-
-      const finalRecordsMatched = fakeFrames[9].meta!.stats!.find(
-        (stat) => stat.displayName === 'Records scanned'
-      )!.value!;
-      for (let i = 10; i < fakeFrames.length; i++) {
-        fakeFrames[i].meta!.stats = [
-          {
-            displayName: 'Records scanned',
-            value: finalRecordsMatched,
-          },
-        ];
-      }
-
-      let i = 0;
-      jest.spyOn(runner, 'makeLogActionRequest').mockImplementation((subtype: LogAction) => {
-        if (subtype === 'GetQueryResults') {
-          const mockObservable = of([fakeFrames[i]]);
-          i++;
-          return mockObservable;
-        } else {
-          return of([]);
-        }
+      const options: DataQueryRequest<CloudWatchLogsQuery> = {
+        ...LogsRequestMock,
+        targets: rawLogQueriesStub,
+      };
+
+      const queryFn = jest
+        .fn()
+        .mockReturnValueOnce(of(startQuerySuccessResponseStub))
+        .mockReturnValueOnce(of(getQuerySuccessResponseStub));
+
+      const response = runner.handleLogQueries(
+        rawLogQueriesStub,
+        options,
+        queryFn as (request: DataQueryRequest<CloudWatchQuery>) => Observable<DataQueryResponse>
```

Comment:
You can get rid of the type casting of `queryFn` for all of the tests in this file by changing

```ts
      const queryFn = jest
        .fn()
```

to

```ts
      const queryFn = jest
        .fn<Observable<DataQueryResponse>, [DataQueryRequest<CloudWatchQuery>]>()
```

And in the response stubs change the `state` from a string to use the `LoadingState` enum from `@grafana/data`